lightning/src/ln/payment_tests.rs

```diff
@@ -1385,3 +1386,74 @@ fn abandoned_send_payment_idempotent() {
 	pass_along_route(&nodes[0], &[&[&nodes[1]]], 100_000, second_payment_hash, second_payment_secret);
 	claim_payment(&nodes[0], &[&nodes[1]], second_payment_preimage);
 }
+
+
+#[test]
+fn test_trivial_inflight_htlc_tracking() {
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	let (_, _, chan_id, _) = create_announced_chan_between_nodes(&nodes, 0, 1, channelmanager::provided_init_features(), channelmanager::provided_init_features());
+
+	// Send and claim the payment. Inflight HTLCs should be empty.
+	send_payment(&nodes[0], &vec!(&nodes[1])[..], 500000);
+	assert_eq!(node_chanmgrs[0].compute_inflight_htlcs().0.len(), 0);
+
+	// Send the payment, but do not claim it. Our inflight HTLCs should contain the pending payment
+	route_payment(&nodes[0], &vec!(&nodes[1])[..], 500000);
```

Comment:
Could you complete the payment after the below assertions and then check that the in-flight HTLCs are empty?